api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1505,4 +1608,23 @@ public void changePasswordUsingActivationKey_shouldNotUpdatePasswordIfActivation
 		
 		userService.changePasswordUsingActivationKey(key, "Pa55w0rd");
 	}
+
+	/**
+	 * Utility method to set the current executing user to test various permission levels.
+	 * 
+	 * @param user the user to set as the currently running user
+	 * @param internals the functionality to test with this current running user
+	 * @throws IllegalAccessException because we use reflection to set the currently running user, this may fail and throw
+	 *  and {@link IllegalArgumentException}
+	 */
+	private void withCurrentUserAs(User user, Runnable internals) throws IllegalAccessException {
+		UserContext userContext = Context.getUserContext();
+		User authenticatedUser = userContext.getAuthenticatedUser();
+		try {
+			FieldUtils.getField(UserContext.class, "user", true).set(userContext, user);
+			internals.run();
+		} finally {
+			FieldUtils.getField(UserContext.class, "user", true).set(userContext, authenticatedUser);
+		}
+	}
```

Comment:
This shouldn't be in this PR

Reply:
Have fixed them , everything here new came due to new commits after rebasing it, so they dont have any harm